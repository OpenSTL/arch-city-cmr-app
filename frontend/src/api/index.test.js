const api = require('./index');

describe('api', () => {
    describe('createApplicant', () => {
        it('should fetch from api/applicants/add', async () => {
            const mockSuccessResponse = {};
            const mockJsonPromise = Promise.resolve(mockSuccessResponse);
            const mockFetchPromise = Promise.resolve({
                json: () => mockJsonPromise,
            });
            jest.spyOn(global, 'fetch').mockImplementationOnce(() => mockFetchPromise);

            const result = await api.createApplicant({}, 'mytoken');
            expect(global.fetch).toHaveBeenCalledTimes(1);
            expect(global.fetch.mock.calls[0][0]).toEqual('http://test/applicants/add');
            expect(global.fetch.mock.calls[0][1]).toMatchSnapshot();
            expect(result).toEqual(mockSuccessResponse);
        });
    });
});
