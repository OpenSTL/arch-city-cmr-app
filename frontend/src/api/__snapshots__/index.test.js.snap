// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api createApplicant should fetch from api/applicants/add 1`] = `
Object {
  "body": "{}",
  "headers": Object {
    "Authorization": "Bearer mytoken",
  },
  "method": "POST",
}
`;
